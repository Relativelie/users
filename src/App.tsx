import './App.scss';

function App() {
    return (
        <div className="App">
            fhfhfh
        </div>
    );
}

export default App;
