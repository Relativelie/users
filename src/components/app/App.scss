body {
    margin: 0;
}

.App {
    display: flex;
    justify-content: center;
}

.container {
    width: 665px;
    display: grid;
    grid-template-areas: "menu listElements";
    grid-template-columns: 152px 1fr;
    height: 100vh;
}

.container__info {
    padding: 19px 59px 0 34px;
    overflow: auto;
}