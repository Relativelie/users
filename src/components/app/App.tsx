import { ListsOfUsers } from '../ListsOfUsers/ListsOfUsers';
import { Filter } from '../filter/Filter';
import { UserInfo } from '../userInfo/UserInfo';
import { useTypedSelector } from '../../hooks/useTypedSelector';
import './App.scss';

function App() {
    const { isOpenListCard } = useTypedSelector(
        (listsOfUsersState) => listsOfUsersState.listOfUsersReducer,
    );

    return (
        <div className="App">
            <div className="container">
                <Filter />
                <main className="container__info">
                    {isOpenListCard ? <UserInfo /> : <ListsOfUsers />}
                </main>

            </div>
        </div>
    );
}

export default App;
