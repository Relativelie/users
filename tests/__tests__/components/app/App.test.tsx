import App from '../../../../src/components/app/App';
import { render } from '../../../testsSetup/test-utils';

describe('App component', () => {
    test('snapshot - app component', () => {
        const tree = render(<App />);
        expect(tree).toMatchSnapshot();
    });
});
