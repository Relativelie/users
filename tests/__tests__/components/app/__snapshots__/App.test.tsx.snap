// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot - app component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <div
          class="container"
        >
          <section
            class="menuBlock"
          >
            <div
              class="menuBlock__filter"
            >
              <h3
                class="filter__header"
              >
                Сортировка
              </h3>
              <button
                class="filter__button "
                data-filterby="address_city"
                type="button"
              >
                по городу
              </button>
              <button
                class="filter__button "
                data-filterby="company_name"
                type="button"
              >
                по компании
              </button>
            </div>
          </section>
          <main
            class="container__info"
          >
            <section
              class="list"
            >
              <h3
                class="list__header"
              >
                Список пользователей
              </h3>
              <div
                class="list__cards"
              >
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Leanne Graham
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      Gwenborough
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Romaguera-Crona
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/1"
                  >
                    Подробнее
                  </a>
                </div>
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Ervin Howell
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      Wisokyburgh
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Deckow-Crist
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/2"
                  >
                    Подробнее
                  </a>
                </div>
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Clementine Bauch
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      McKenziehaven
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Romaguera-Jacobson
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/3"
                  >
                    Подробнее
                  </a>
                </div>
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Patricia Lebsack
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      South Elvis
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Robel-Corkery
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/4"
                  >
                    Подробнее
                  </a>
                </div>
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Chelsey Dietrich
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      Roscoeview
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Keebler LLC
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/5"
                  >
                    Подробнее
                  </a>
                </div>
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Mrs. Dennis Schulist
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      South Christy
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Considine-Lockman
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/6"
                  >
                    Подробнее
                  </a>
                </div>
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Kurtis Weissnat
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      Howemouth
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Johns Group
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/7"
                  >
                    Подробнее
                  </a>
                </div>
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Nicholas Runolfsdottir V
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      Aliyaview
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Abernathy Group
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/8"
                  >
                    Подробнее
                  </a>
                </div>
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Glenna Reichert
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      Bartholomebury
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Yost and Sons
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/9"
                  >
                    Подробнее
                  </a>
                </div>
                <div
                  class="list__card"
                >
                  <div>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        ФИО:
                      </span>
                      Clementina DuBuque
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        город:
                      </span>
                      Lebsackbury
                    </p>
                    <p
                      class="list__cardContent"
                    >
                      <span
                        class="list__cardContentTitle"
                      >
                        компания:
                      </span>
                      Hoeger LLC
                    </p>
                  </div>
                  <a
                    class="list__more"
                    href="/user/10"
                  >
                    Подробнее
                  </a>
                </div>
              </div>
              <p
                class="list__resultCount"
              >
                Найдено 10 пользователей
              </p>
            </section>
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <div
        class="container"
      >
        <section
          class="menuBlock"
        >
          <div
            class="menuBlock__filter"
          >
            <h3
              class="filter__header"
            >
              Сортировка
            </h3>
            <button
              class="filter__button "
              data-filterby="address_city"
              type="button"
            >
              по городу
            </button>
            <button
              class="filter__button "
              data-filterby="company_name"
              type="button"
            >
              по компании
            </button>
          </div>
        </section>
        <main
          class="container__info"
        >
          <section
            class="list"
          >
            <h3
              class="list__header"
            >
              Список пользователей
            </h3>
            <div
              class="list__cards"
            >
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Leanne Graham
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    Gwenborough
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Romaguera-Crona
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/1"
                >
                  Подробнее
                </a>
              </div>
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Ervin Howell
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    Wisokyburgh
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Deckow-Crist
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/2"
                >
                  Подробнее
                </a>
              </div>
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Clementine Bauch
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    McKenziehaven
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Romaguera-Jacobson
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/3"
                >
                  Подробнее
                </a>
              </div>
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Patricia Lebsack
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    South Elvis
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Robel-Corkery
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/4"
                >
                  Подробнее
                </a>
              </div>
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Chelsey Dietrich
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    Roscoeview
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Keebler LLC
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/5"
                >
                  Подробнее
                </a>
              </div>
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Mrs. Dennis Schulist
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    South Christy
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Considine-Lockman
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/6"
                >
                  Подробнее
                </a>
              </div>
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Kurtis Weissnat
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    Howemouth
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Johns Group
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/7"
                >
                  Подробнее
                </a>
              </div>
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Nicholas Runolfsdottir V
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    Aliyaview
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Abernathy Group
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/8"
                >
                  Подробнее
                </a>
              </div>
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Glenna Reichert
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    Bartholomebury
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Yost and Sons
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/9"
                >
                  Подробнее
                </a>
              </div>
              <div
                class="list__card"
              >
                <div>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      ФИО:
                    </span>
                    Clementina DuBuque
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      город:
                    </span>
                    Lebsackbury
                  </p>
                  <p
                    class="list__cardContent"
                  >
                    <span
                      class="list__cardContentTitle"
                    >
                      компания:
                    </span>
                    Hoeger LLC
                  </p>
                </div>
                <a
                  class="list__more"
                  href="/user/10"
                >
                  Подробнее
                </a>
              </div>
            </div>
            <p
              class="list__resultCount"
            >
              Найдено 10 пользователей
            </p>
          </section>
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
